app/commands/create_grade_year.rb

```diff
@@ -0,0 +1,25 @@
+class CreateGradeYear < ROM::Commands::Create[:sql]
+  relation :grade_years
+  register_as :create
+  result :one
+
+  def execute(attributes)
+    gpa = attributes.delete(:gpa).to_f
+    if gpa > 3
+      letter = 'A'
+    elsif gpa > 1
+      letter = 'D'
+    end
```

Comment:
This doesn't really belong here. Command should simply receive attributes. If you want to associate something you can use `associates :grade, key: [:grade_id, :id]` macro and pass `grade` with its `:id` to command along with grade year attributes.